src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -483,72 +483,100 @@ quota_t cap_to_ceiling(quota_t& value, const quota_t limit) {
     return 0;
 }
 
-/// If \p delta is negative, splits its value accourding to \p surplus values:
-/// - amount under total surplus is split pro rata to \p surplus amounts
-/// - amount over total surplus is split equally. Values in \p schedule
-/// are added to.
-/// \pre schedule.size() == surplus.size()
+/// If \p delta is negative, splits its value pro rata to \p quotas values and
+/// into the slices limited by \p quota values. The slices are added to the
+/// elements of \p schedule.
+/// \pre sum(\p quotas) < abs(delta)
+/// \pre schedule.size() == quotas.size()
+/// \pre 0 <= quotas[i] for any i
 void dispense_negative_deltas(
   std::vector<quota_t>& schedule,
-  quota_t& delta,
-  const std::vector<quota_t>& surplus) {
+  quota_t delta,
+  std::vector<quota_t>&& quotas) {
```

Comment:
pass by value, unless you're modifying it for the caller. in the later case it should be an l-value reference.

Reply:
then it will be a copy of a vector, although small but it's still an avoidable allocation. IMO this pattern (take arg by rvalue ref) expresses the idea that the argument is handed over into the function and is not used at the call site afterwards, and also that it's only an 'in' argument, not an 'in,out'.

Reply:
>then it will be a copy of a vector, although small but it's still an avoidable allocation.

there are no copies or allocations if you std::move the vector into the parameter as a value.

Reply:
> if you std::move the vector into the parameter value, or pass a temporary

I keep forgetting about that. Totally makes sense.